Installation.md

```diff
@@ -0,0 +1,267 @@
+## Setup ##
+
+### Notices before installation ###
+
+We assume you use [RVM](https://rvm.io/). If you don't, then replace `rvmsudo` with `sudo` during the installation process.
+
+:exclamation: RVM 1.17.x may have some [problems with rvmsudo](http://stackoverflow.com/questions/13765520/rvmsudo-command-not-working-properly).
+
+An X server is currently needed for browser tests to pass when using Selenium. `Xvfb` will do, but remember to set your `DISPLAY`.
+
+:exclamation: For testing environment PhantomJS (at least version 1.8.1) is required by Poltergeist gem to run headless browser tests. See [poltergeist](https://github.com/teampoltergeist/poltergeist) for documentation.
+
+Below is rough setup instructions, if you're running Ubuntu you can follow the [Ubuntu specific installation guide](#installation).
+
+### System Dependencies ###
+
+The following programs should be installed first: `ruby` (and `bundler` gem), `PostgreSQL 9.2+`, `git`, `zip`, `unzip`, `convert` (from ImageMagick), `javac`, `java`, `ant`, `mvn`, `gcc`, `make` and `bc`. As well as tmc-sandbox dependencies: `squashfs-tools`, `multistrap` `e2fsprogs`, `e2tools` and `build-essential`.
+
+1. Download submodules with `git submodule update --init --recursive`
+2. Install dependencies with `bundle install`
+3. Edit `config/site.yml`, based on `config/site.defaults.yml`.
+4. Setup PostgreSQL based on `config/database.yml` or configure the file according to your current postgres settings.
+5. Initialize rails database and precompile assets with `env RAILS_ENV=production rake db:reset` and `env RAILS_ENV=production rake assets:precompile`
+6. Go to `ext/tmc-sandbox` and compile with `sudo make`. More information from [tmc-sandbox page](https://github.com/testmycode/tmc-sandbox).
+7. Go to `ext/tmc-sandbox/web` and install dependencies with `bundle install`. Compile extensions with `rake ext` and run tests with `rvmsudo rake test`.
+8. Compile the other stuff in `ext` by doing `rake compile`
+9. Install [tmc-check](https://github.com/testmycode/tmc-check) by running `rvmsudo make rubygems install clean` in `ext/tmc-sandbox/uml/output/tmc-check`.
+10. Run the test suite with `rvmsudo rake spec`.
+11. Get pghero stats enabled, add to postgresql.conf following lines `shared_preload_libraries = 'pg_stat_statements'` and `pg_stat_statements.track = all`
+
+After you get the test suite to pas you can set up [background services](#background-services) and [startup](#startup) the server.
+
+#### Installation ####
```

Comment:
Should be h3, not h4.

Please add: `The following is a set of detailed instructions for setting tmc-server up on Ubuntu 14.04.`